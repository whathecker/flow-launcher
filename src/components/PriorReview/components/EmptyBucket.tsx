import React from "react";
import { StyleSheet } from "react-native";
import { View, Text } from "../../Themed";
import { Container, Typography, Color } from "../../../styles";

type EmptyBucketProps = {
  title: string;
};

const EmptyBucket: React.FC<EmptyBucketProps> = ({
  title,
}: EmptyBucketProps) => {
  return (
    <>
      <Text
        lightColor={Color.light.labelOnBackgroundForRead}
        darkColor={Color.dark.labelOnBackgroundForRead}
        style={styles.header}
      >
        {title}
      </Text>
      <View
        lightColor={Color.light.emptyPrioBucket}
        darkColor={Color.dark.emptyPrioBucket}
        style={styles.innerWrapper}
      >
        <Text
          lightColor={Color.light.labelOnBackgroundForRead}
          darkColor={Color.dark.labelOnBackgroundForRead}
          style={styles.placeholderText}
        >{`No awaiting tasks`}</Text>
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  header: {
    ...Typography.h4,
    fontSize: 16,
    paddingLeft: "5%",
    paddingTop: 5,
  },
  innerWrapper: {
    ...Container.centerAligned,
    paddingTop: "12%",
    paddingBottom: "6%",
    backgroundColor: "#FEFEF8",
  },
  placeholderText: {
    ...Typography.p,
    fontSize: 14,
  },
});

export default EmptyBucket;
