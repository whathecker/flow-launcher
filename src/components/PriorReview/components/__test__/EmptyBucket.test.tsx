import * as React from "react";
import renderer from "react-test-renderer";
import EmptyBucket from "../EmptyBucket";

describe("Test EmptyBucket component", () => {
  it("Should renders EmptyBucket correctly", () => {
    const tree = renderer
      .create(<EmptyBucket title="This should be title of empty prio bucket" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
