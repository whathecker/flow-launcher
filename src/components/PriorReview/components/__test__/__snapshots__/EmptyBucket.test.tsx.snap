// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test EmptyBucket component Should renders EmptyBucket correctly 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#848484",
        },
        Object {
          "fontFamily": "Lexend-SemiBold",
          "fontSize": 16,
          "paddingLeft": "5%",
          "paddingTop": 5,
        },
      ]
    }
  >
    This should be title of empty prio bucket
  </Text>,
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FEFEF8",
        },
        Object {
          "alignItems": "center",
          "backgroundColor": "#FEFEF8",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": "6%",
          "paddingTop": "12%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#848484",
          },
          Object {
            "fontFamily": "Lexend-Regular",
            "fontSize": 14,
          },
        ]
      }
    >
      No awaiting tasks
    </Text>
  </View>,
]
`;
